runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -413,6 +428,114 @@ class VM_JFRConstantPoolTypes {
 		return;
 	}
 
+	const char *getSystemProp(const char *propName) {
+		J9VMSystemProperty *jvmInfoProperty = NULL;
+		J9InternalVMFunctions *vmFuncs = _vm->internalVMFunctions;
+		const char *value = "";
+		UDATA getPropertyResult = vmFuncs->getSystemProperty(_vm, propName, &jvmInfoProperty);
+		if (J9SYSPROP_ERROR_NOT_FOUND != getPropertyResult) {
+			value = jvmInfoProperty->value;
+		}
+		return value;
+	}
+
+	J9UTF8 *makeJ9UTF8(std::initializer_list<std::pair<const char *, UDATA>> parts) {
+		UDATA len = 0;
+
+		for (auto s = parts.begin(); s != parts.end(); ++s) {
+			len += s->second;
+		}
+
+		J9UTF8 *result = (J9UTF8 *)j9mem_allocate_memory(sizeof(J9UTF8) + len, OMRMEM_CATEGORY_VM);
+		U_8 *data = J9UTF8_DATA(result);
+
+		for (auto s = parts.begin(); s != parts.end(); ++s) {
+			memcpy(data, s->first, s->second);
+			data += s->second;
+		}
+
+		J9UTF8_SET_LENGTH(result, len);
+
+		return result;
+	}
+
+    void initJVMInfo() {
+		const char *javaVMName = getSystemProp("java.vm.name");
+		const UDATA javaVMNameLen = strlen(javaVMName);
+
+		const char *jdkDebugLevel = getSystemProp("jdk.debug");
+		const UDATA jdkDebugLevelLen = strlen(jdkDebugLevel);
+
+		const char *javaVMVersion = getSystemProp("java.vm.version");
+		const UDATA javaVMVersionLen = strlen(javaVMVersion);
+
+		const char *javaVMInfo = getSystemProp("java.vm.info");
+		const char *newLineChar = strchr(javaVMInfo, '\n'); /* only keep the first line */
+		const UDATA javaVMInfoLen = (NULL == newLineChar ? strlen(javaVMInfo) : newLineChar - javaVMInfo);
+
+		_jvmInformation = {0};
+		/* Set jvmName */
+		_jvmInformation.jvmName = makeJ9UTF8({{javaVMName, javaVMNameLen}});
+
+		/* Set jvmVersion */
+		_jvmInformation.jvmVersion = makeJ9UTF8({
+			{ javaVMName, javaVMNameLen },
+			{ " (", 2 },
+			{ jdkDebugLevel, jdkDebugLevelLen },
+			{ "build ", 6 },
+			{ javaVMVersion, javaVMVersionLen },
+			{ ", ", 2 },
+			{ javaVMInfo, javaVMInfoLen },
+			{ ")", 1 },
+		});
+
+		/* Set JVM arguments */
+		JavaVMInitArgs* vmArgs = _vm->vmArgsArray->actualVMArgs;
+		const char *javaCommand = "-Dsun.java.command=";
```

Comment:
since `-Dsun.java.command=` looks something like `[javaAppName] [app args ...]`, you can use `strchr` to find the first space, the return that pointer

Reply:
RI includes `javaAppName` as part of Java arguments.

Reply:
In that case you can keep it